### PR TITLE
[MM-21919] Add use_channel_mentions permission

### PR DESCRIPTION
**Pull request by fm2munsh**
#### Summary
- Adds a migration to add the new `channel_mention` permission, gives it to any role that has `create_post` or `create_post_public` so that users / bots that previously had the ability to channel mention will still have the ability channel mention when we start enforcing this permission.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21919

### CODE REVIEW

**Comment by fm2munsh** [comment]
Yea - I am planning on actually enforcing the permission in a different PR built off of this

**Comment by fm2munsh** [comment]
Actually @jespino I decided to enforce server side checks for the permission in this same PR and I will do the client side checks as part of different tickets 

I realized there was no ticket for enforcing server side checks for the new permission so it just made sense to add it to this since it was simple enough instead of making a new ticket / PR

File: app/permissions_migrations.go [hunk 1]
```diff
@@ -56,6 +57,9 @@ const (
 	PERMISSION_INVITE_GUEST                      = "invite_guest"
 	PERMISSION_PROMOTE_GUEST                     = "promote_guest"
 	PERMISSION_DEMOTE_TO_GUEST                   = "demote_to_guest"
+	PERMISSION_CHANNEL_MENTION                   = "channel_mention"
```
Comment (mkraft) [inline app/permissions_migrations.go]: All of the other permissions contain a verb but this one does not. I'm wondering if `use_channel_mentions` is more idiomatic. I suppose the argument could be made that "mention" is the verb, and in that case would `mention_channel` be better?

Comment (fm2munsh) [reply app/permissions_migrations.go]: I think `mention_channel` makes the most sense - like you said `mention` is the verb like `delete_channel` or `create_channel`.

Comment (fm2munsh) [reply app/permissions_migrations.go]: actually I just got went through the tech spec again and it actually has the permission down as `use_channel_mentions` so ill go with that.

Comment (mkraft) [reply app/permissions_migrations.go]: @fm2munsh Oops, I just changed it. It's your call I can update the spec to fit your decision.

Comment (fm2munsh) [reply app/permissions_migrations.go]: haha I just modified my code to `use_channel_mentions` so lets revert it back to that if that still works for you 🙂

Comment (mkraft) [reply app/permissions_migrations.go]: @fm2munsh Yeah sounds good; will do 👍

**Comment by justinegeffen** [comment]
This is included in https://github.com/mattermost/docs/pull/3496. No additional documentation needed.

**Pull request merged**